src/hpr_routing.erl

```diff
@@ -97,95 +103,159 @@ find_routes({uplink, {_Type, DevAddr}}, PacketUp) ->
     ok = hpr_metrics:observe_find_routes(Time),
     Results.
 
+-spec start_crawl_routing() -> pid().
+start_crawl_routing() ->
+    erlang:spawn(fun() ->
+        Timer = timer:seconds(hpr_utils:get_env_int(routing_cleanup_timer_secs, 10)),
+        Window = timer:seconds(hpr_utils:get_env_int(routing_cleanup_window_secs, 60)),
+        ok = timer:sleep(Timer),
+        Removed = do_crawl_routing(Window),
+        lager:debug(
+            [{timer, Timer}, {window, Window}, {removed, Removed}],
+            "cleaning routing table"
+        ),
+        true = erlang:unregister(?ROUTING_CLEANUP),
+        true = erlang:register(?ROUTING_CLEANUP, start_crawl_routing())
+    end).
+
+-spec cancel_crawl_routing() -> ok.
+cancel_crawl_routing() ->
+    true = exit(get_crawl_routing_pid(), kill),
```

Comment:
Again, typically this would be a cast into gen_server state rather than an explicit message passed to the process

Reply:
Yes, to echo Jeff & Jade's comments, I think  it might be easier to create a quick `gen_server` (`hpr_routing_cache` ?) that can handle that ETS table as well as the cleanup and all

Reply:
I would also make it a record instead of a 3 tuple in the ETS, easier to read IMO.

Reply:
moved to `hpr_routing_cache` gen_server. and a record.